File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        for(String[] cloth : clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 0) + 1);
        }
        int sum = 1;
        for(String key : map.keySet()) {
            sum *= map.get(key) + 1;
        }
        return sum - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> map;
    for(vector<string> cloth : clothes) {
        map[cloth[1]]++;
    }
    
    int sum = 1;
    for(auto m : map) {
        sum *= (m.second + 1);
    }
    return sum - 1;
}